database/wireless_regdb.ksy

```diff
@@ -0,0 +1,196 @@
+  Copyright (c) 2008, Johannes Berg <johannes@sipsolutions.net>
+  Copyright (c) 2008, Michael Green <Michael.Green@Atheros.com>
+
+  Permission to use, copy, modify, and/or distribute this software for any purpose with or without fee is hereby granted, provided that the above copyright notice and this permission notice appear in all copies.
+
+  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+doc: |
+  the test file: https://git.kernel.org/pub/scm/linux/kernel/git/sforshee/wireless-regdb.git/plain/regulatory.bin
+doc-ref:
+  - "https://wireless.wiki.kernel.org/en/developers/Regulatory/wireless-regdb"
+  - "https://git.kernel.org/pub/scm/linux/kernel/git/mcgrof/crda.git/tree/regdb.h"
+  - "https://git.kernel.org/pub/scm/linux/kernel/git/mcgrof/crda.git/tree/reglib.c"
+  - "https://git.kernel.org/pub/scm/linux/kernel/git/sforshee/wireless-regdb.git/tree/db2bin.py"
+seq:
+  - id: signature
+    contents: ["RGDB"]
+  - id: version
+    type: u4
+  - id: realms_ptr
+    -orig-id: countries_ptr
+    type: u4
+    doc: The list is sorted alphabetically to allow binary searching (should it become really huge).
+  - id: realms_count
+    -orig-id: countries_count
+    type: u4
+  - id: authentication_size
+    -orig-id: signature_len
+    type: u4
+    doc: length (in bytes) of the signature at the end of the file
+instances:
+  realms:
+    pos: realms_ptr
+    type: realm
+    repeat: expr
+    repeat-expr: realms_count
+  authentication:
+    pos: _io.size - authentication_size
+    size: authentication_size
+    doc: digital signature
+types:
+  freq_range:
+    doc: |
+      everything is in in kHz
+      Values of zero mean "not applicable", i.e. the regulatory does not limit a certain value.
+    seq:
+     - id: start
+       type: u4
+     - id: end
+       type: u4
+     - id: max_bandwidth
+       type: u4
+  power_rule:
+    seq:
+      - id: max_antenna_gain
+        type: u4
+        doc: in mBi (100 * dBi)
+      - id: max_eirp
+        type: u4
+        doc: in mBi (100 * dBi)
+  rule:
+    seq:
+      - id: freq_range_ptr
+        type: u4
+      - id: power_rule_ptr
+        type: u4
+      - id: flags
+        type: flags
+    instances:
+      power_rule:
+        pos: power_rule_ptr
+        type: power_rule
+      freq_range:
+        pos: freq_range_ptr
+        type: freq_range
+  rule_ptr:
+    seq:
+      - id: ptr
+        type: u4
+    instances:
+      rule:
+        pos: ptr
+        type: rule
+  flags:
+    seq:
+      - id: reserved1
+        type: b20
```

Comment:
Given that there are bit-sized integers used here, there should be explicit `meta/bit-endian: be` at the top level. See https://doc.kaitai.io/user_guide.html#bit-endian.

Reply:
Fixed, thanks.